app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationActivityV2.kt

```diff
@@ -1835,10 +1904,23 @@ class ConversationActivityV2 : PassphraseRequiredActionBarActivity(), InputBarDe
     }
 
     override fun cancelVoiceMessage() {
+        val inputBar = binding?.inputBar!!
+
         hideVoiceMessageUI()
         window.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
         audioRecorder.stopRecording()
         stopAudioHandler.removeCallbacks(stopVoiceMessageRecordingTask)
+
+        // Note: The 0L check prevents the warning toast being shown when leaving the conversation activity
+        val voiceMessageDuration = inputBar.voiceMessageDurationMS
+        if (voiceMessageDuration != 0L && voiceMessageDuration < MinimumVoiceMessageDurationMS) {
+            Toast.makeText(applicationContext, applicationContext.getString(R.string.messageVoiceErrorShort), Toast.LENGTH_SHORT).show()
```

Comment:
Is there an reason this is defined differently than what you are doing in `sendVoiceMessage`?

Reply:
I don't believe there's anything different about the few lines of code that show the toast here vs. in `sendVoiceMessage`, is there?